deploy-tool/cert/cert.go

```diff
@@ -0,0 +1,90 @@
+package cert
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/x509"
+	"deploy-tool/config"
+	"deploy-tool/dns"
+	"fmt"
+	"github.com/caddyserver/certmagic"
+	"strings"
+)
+
+// IssueCertificate asks the Let's Encrypt ACMEApiEndpoint to validate and sign a certificate for a site using the DNS01 strategy
+func IssueCertificate(deployConfig config.DeployConfig, site string, privKey *ecdsa.PrivateKey) (string, error) {
+	if deployConfig.ACME.SubscriberAgreement == false {
+		return "", fmt.Errorf("let's encrypt ACMEApiEndpoint terms were not agreed to")
+	}
```

Comment:
Why not just set `certmagic.DefaultACME.Agreed = deployConfig.ACME.SubscriberAgreement` below, and then let Boulder or CertMagic throw errors if they don't like the value? Once again, it would be less code and fewer test cases for us to worry about.